easystroke: init at unstable-2016-07-05

ivan opened the pull request:
###### Motivation for this change

[easystroke](https://github.com/thjaeger/easystroke) allows binding mouse gestures to commands. Upstream development has apparently ceased, but the program is very useful and, as far as I know, there is no equivalent alternative.

@revast [investigated which patches should be applied](https://github.com/thjaeger/easystroke/pull/10#issuecomment-444132355) and I've closely followed [the list of patches in the new build instructions](https://github.com/thjaeger/easystroke/wiki/BuildInstructions#releases).

I have tested the resulting binary and it seems to work fine with my existing configuration and without an existing configuration.

This PR obsoletes #37353.  I have fixed the things @andir mentioned in the review.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

ivan committed:
easystroke: init at unstable-2016-07-05

This commit is based on work by Andreas Schaefer in
https://github.com/NixOS/nixpkgs/pull/37353

ivan commented:
Updated to use `dbus-glib` instead of `dbus_glib` to hopefully make grahamcofborg happy.

revast commented:
nice! Thanks for packaging!

jtojnar commented on pkgs/applications/misc/easystroke/default.nix:
```diff
@@ -0,0 +1,70 @@
+
+  nativeBuildInputs = [ help2man pkgconfig wrapGAppsHook ];
+  buildInputs = [
+    boost dbus-glib gnome3.defaultIconTheme gnome3.gtkmm intltool
```
`intltool` belongs to `nativeBuildInputs`

jtojnar commented on pkgs/applications/misc/easystroke/default.nix:
```diff
@@ -0,0 +1,70 @@
+  ];
+
+  installPhase = ''
+    make install PREFIX=$out
```
You can use `installFlags` or `makeFlags` instead of overriding the whole command and forgetting to run pre/post hooks.

jtojnar commented on pkgs/applications/misc/easystroke/default.nix:
```diff
@@ -0,0 +1,70 @@
+    })
+    (fetchpatch {
+      # Add tray icon option to toggle easystroke on/off
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/add-toggle-option.patch?h=easystroke-git";
```
You ought to pin this to a specific commit revision, in order to prevent breakage:

https://aur.archlinux.org/cgit/aur.git/plain/add-toggle-option.patch?h=easystroke-git&id=fcd9c1080cddbabae3e85a2f6d8c8be5c2b7c13d

jtojnar commented on pkgs/applications/misc/easystroke/default.nix:
```diff
@@ -0,0 +1,70 @@
+
+    (fetchpatch {
+      # Fix build failure with libsignc++ version 2.5.1 or newer
+      url = "https://github.com/thjaeger/easystroke/pull/9/commits/22b28d25bb696e37e73b4bc641439b3db9f564ed.patch";
```
Please change the URL to https://github.com/thjaeger/easystroke/commit/22b28d25bb696e37e73b4bc641439b3db9f564ed.patch in case the PR gets rebased and the ref dropped (not certain if this can happen).

jtojnar commented on pkgs/applications/misc/easystroke/default.nix:
```diff
@@ -0,0 +1,70 @@
+    })
+    (fetchpatch {
+      # Remove abs(float) function that clashes with std::abs(float)
+      url = "https://github.com/thjaeger/easystroke/commit/9e2c32390c5c253aade3bb703e51841748d2c37e.patch";
```
No need to quote the links.

ivan commented:
Thanks for the review!  I think I fixed everything mentioned.  I was also required to add two `name = `s for the patches that now have a `&` in their URLs.

I also noticed that the man page wasn't being installed, so I fixed that as well.

ivan commented:
Updated to add `knownVulnerabilities` (not sure if even suitable for inclusion now?), then updated to replace `xorg.inputproto` (now gone) with `xorg.xorgproto`.

winpat commented:
@jtojnar I would be very much interested in this.

ivan commented:
I am currently testing my fork https://github.com/ludios/easystroke to see if the rare Xorg crash (when starting a stroke) is fixed by [this change](https://github.com/ludios/easystroke/commit/63b13f46c48dd7146df131e88f86b9bf0d850c70), and  I will update this PR if it does.

nixos-discourse commented:
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/26

ivan commented:
I wasn't able to resolve the occasional Xorg crash-on-stroke issue in my fork, and while I continue to use easystroke, I suggest avoiding it unless someone fixes the crasher. Anyone else is welcome to pick this up if they wish.
